src/main/java/io/quarkus/registry/app/model/PlatformRelease.java

```diff
@@ -203,14 +207,12 @@ public static List<PlatformRelease> findLatest(String quarkusCore) {
         for (var item : query.getResultList()) {
             var platformReleases = platformMap.computeIfAbsent(item.platformStream.platform,
                     k -> new ArrayList<>());
-            if (platformReleases.size() < 3) {
-                // Count stable versions
-                long stableVersions = platformReleases.stream()
-                        .filter(pr -> !Version.isPreFinal(pr.version))
-                        .count();
-                if (stableVersions < 2) {
-                    platformReleases.add(item);
-                }
+            // Count stable versions
+            long stableVersions = platformReleases.stream()
+                    .filter(PlatformRelease::isVersionStable)
+                    .count();
+            if (stableVersions == 0) {
+                platformReleases.add(item);
```

Comment:
I'm confused, tbh. How do stable versions get added? Also looks like the commit above should be updated.

Reply:
Comment updated. Because the query returned will be ordered by the most recent version, in case the latest is not a stable version, it will include all the non-stable versions until the next stable is found.

Reply:
It looks to me as only non-stable and pinned versions are collected. How stable versions are added?

Reply:
The `item` in the loop may be a stable version (it represents the latest and greatest release per stream/platform)

Reply:
Only the first (the greatest version) will be added

Reply:
Ok, I see it now, thanks!